test/1_test_fullpath.js

```diff
@@ -184,37 +184,43 @@ contract('Voucher tests', async (addresses) => {
 
       //Check BosonRouter state
       assert.equal(
-        await contractBosonRouter.correlationIds(users.seller.address),
+        await contractBosonRouter.getCorrelationId(users.seller.address),
         1,
         'Correlation Id incorrect'
       );
 
       //Check VocherKernel State
-      const promise = await contractVoucherKernel.promises(promiseId1);
-      assert.equal(promise.promiseId, promiseId1, 'Promise Id incorrect');
-      assert.isTrue(promise.nonce.eq(constants.ONE), 'Nonce is incorrect');
+      const promiseData = await contractVoucherKernel.getPromiseData(
+        promiseId1
+      );
+      assert.equal(promiseData[0], promiseId1, 'Promise Id incorrect');
```

Comment:
please define constants instead of hard-coded indexes (0,1,2,3,...) for the PromiseData.
For instance:
```
      assert.equal(promiseData[PROMISE_DATA_FIELDS.promiseId], promiseId1, 'Promise Id incorrect');
      assert.isTrue(promiseData[PROMISE_DATA_FIELDS.nonce].eq(constants.ONE), 'Nonce is incorrect');
      assert.isTrue(promiseData[PROMISE_DATA_FIELDS.validFrom].eq(new BN(constants.PROMISE_VALID_FROM)));
      assert.isTrue(promiseData[PROMISE_DATA_FIELDS.validTo].eq(new BN(constants.PROMISE_VALID_TO)));
      assert.isTrue(promiseData[PROMISE_DATA_FIELDS.idx].eq(constants.ZERO));
```
These constants can be defined in testHelpers/utils so that they can be used in different tests.
If there is a change in the contract (adding a new field in the return), the update is easier (and the risk of mistake very lower).
An example here: https://github.com/levalleux-ludo/DeFi-Venture/blob/main/buidler/test/testsUtils.js
(USER_DATA_FIELDS and GAME_DATA_FIELDS) used in that test file:
https://github.com/levalleux-ludo/DeFi-Venture/blob/main/buidler/test/gamemasteroptim-test.js

This remark also apply to other cases in the tests